powermon/ports/serialport.py

```diff
@@ -2,52 +2,51 @@
 import serial
 import time
 
-from .port import Port
-from ..helpers import get_kwargs
+from .abstractport import AbstractPort
 
-log = logging.getLogger("serial")
+log = logging.getLogger("SerialPort")
 
 
-class serialport(Port):
-    def __init__(self, *args, **kwargs) -> None:
-        log.debug(f"Initializing usbserial port args:{args}, kwargs: {kwargs}")
-        self._config = get_kwargs(kwargs, "config")
-        self.port = None
+class SerialPort(AbstractPort):
+    def __init__(self, path, baud, protocol) -> None:
+        log.debug(f"Initializing usbserial port. path:{path}, baud: {baud}")
+        self.path = path
+        self.baud = baud
+        self.protocol = protocol
+        self.serialPort = None
         self.error = None
 
     def connect(self) -> None:
-        log.debug("usbserial port connecting")
-        port = self._config.get("path")
-        baud = self._config.get("baud")
+        log.debug(f"usbserial port connecting. path:{self.path}, baud:{self.baud}")
         try:
-            self.port = serial.Serial(port, baud, timeout=1, write_timeout=1)
+            self.serialPort = serial.Serial(port=self.path, baudrate=self.baud, timeout=1, write_timeout=1)
         except Exception as e:
-            log.warning(f"Error openning serial port: {e}")
+            log.error(f"Error openning serial port: {e}")
             self.error = e
         return
 
     def disconnect(self) -> None:
         log.debug("usbserial port disconnecting")
-        if self.port is not None:
-            self.port.close()
+        if self.serialPort is not None:
+            self.serialPort.close()
         return
 
-    def send_and_receive(self, *args, **kwargs) -> dict:
```

Comment:
wondering if this should use callbacks as the pattern
for normal serial not really a problem, but victron and daly and ble spew info continuously